app/js/filterFunctions.ts

```diff
@@ -129,4 +133,27 @@ function testAllocatedFilter(resultArray: Array<any>) {
         });
     }
     return newResultArray
+}
+
+/**
+ * This function filters the list of users to be displayed in the table based on the category
+ * allocated to that user
+ *
+ * @param {resultArray} Array array of users to be filtered
+ *
+ * @return Array the filtered array of users
+ */
+function categoryFilter(resultArray: Array<any>) {
+    let filterCategory = document.querySelector<HTMLInputElement>('#categoryFilter');
+    let newResultArray = resultArray;
+    let chosenCategory = filterCategory.value;
+    if (chosenCategory) {
+        newResultArray = [];
+        resultArray.forEach( (data) => {
+            if (data.categoryId === chosenCategory) { //need to check that data.category_id is the category id and that it exists in data object
```

Comment:
the comment doesnt feel like it adds anything i cant read from the code

Reply:
removed